src/segment_plastic.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type plastic struct {
+	props properties
+	env   environmentInfo
+
+	Moved    int
+	Deleted  int
+	Added    int
+	Modified int
+	Unmerged int
+	Changed  bool
+
+	Behind   bool
+	Selector string
+
+	plasticWorkspaceFolder string // .plastic working folder
+
+	cmCommand string
+}
+
+const (
+	// FullBranchPath displays the full path of a branch
+	FullBranchPath Property = "full_branch_path"
+)
+
+func (g *plastic) enabled() bool {
+	if !g.env.hasCommand(g.getCmCommand()) {
+		return false
+	}
+	wkdir, err := g.env.hasParentFilePath(".plastic")
+	if err != nil {
+		return false
+	}
+	if g.shouldIgnoreRootRepository(wkdir.parentFolder) {
+		return false
+	}
+
+	if wkdir.isDir {
+		g.plasticWorkspaceFolder = wkdir.parentFolder
+		return true
+	}
+
+	return false
+}
+
+func (g *plastic) shouldIgnoreRootRepository(rootDir string) bool {
+	value, ok := g.props[ExcludeFolders]
+	if !ok {
+		return false
+	}
+	excludedFolders := parseStringArray(value)
+	return dirMatchesOneOf(g.env, rootDir, excludedFolders)
+}
+
+func (g *plastic) string() string {
+	statusColorsEnabled := g.props.getBool(StatusColorsEnabled, false)
+	displayStatus := g.props.getOneOfBool(FetchStatus, DisplayStatus, false)
+
+	g.Selector = g.getSelector()
+	if displayStatus || statusColorsEnabled {
+		g.getPlasticStatus()
+	}
+
+	// use template if available
+	segmentTemplate := g.props.getString(SegmentTemplate, "")
+	if len(segmentTemplate) > 0 {
+		return g.templateString(segmentTemplate)
+	}
+
+	// default: only selector is returned
+	return g.Selector
+}
+
+func (g *plastic) templateString(segmentTemplate string) string {
+	template := &textTemplate{
+		Template: segmentTemplate,
+		Context:  g,
+		Env:      g.env,
+	}
+	text, err := template.render()
+	if err != nil {
+		return err.Error()
+	}
+	return text
+}
+
+func (g *plastic) Status() string {
+	var status string
+	stringIfValue := func(value int, prefix string) string {
+		if value > 0 {
+			return fmt.Sprintf(" %s%d", prefix, value)
+		}
+		return ""
+	}
+	status += stringIfValue(g.Added, "+")
+	status += stringIfValue(g.Modified, "~")
+	status += stringIfValue(g.Deleted, "-")
+	status += stringIfValue(g.Moved, ">")
+	status += stringIfValue(g.Unmerged, "x")
+	return strings.TrimSpace(status)
+}
+
+func (g *plastic) getPlasticStatus() {
+	output := g.getCmCommandOutput("status", "--all", "--machinereadable")
+	splittedOutput := strings.Split(output, "\n")
+	// compare to head
+	currentChangeset := g.parseStatusChangeset(splittedOutput[0])
+	headChangeset := g.getHeadChangeset()
+	g.Behind = headChangeset > currentChangeset
+
+	// parse file state
+	g.parseFilesStatus(splittedOutput)
+}
+
+func (g *plastic) parseFilesStatus(output []string) {
```

Comment:
Same remark as for `gitStatus`, we can make use (extend) the `add` function there

Reply:
I'll have a look at the new `add` logic.
By the way, what does `unmerged` translate to in git? Is it 'files with unhandled merge conflicts' or 'locally merged but uncommited files'? (I implemented the later one for plastic)

Reply:
the first option